dex/testing/dcrdex/harness.sh

```diff
@@ -59,7 +59,25 @@ cat > "./markets.json" <<EOF
 }
 EOF
 
+# Provide the postgres user password if set. 
+PG_PASS=
+
 # Write dcrdex.conf.
+if [ -n "${PG_PASS}" ]; then
+cat > "./dcrdex.conf" <<EOF
+regfeexpub=spubVWHTkHRefqHpw3qMN2VTWqnT9rMuqDZrbdgxp8QzGBvXCYfdUuk2VKFnsPHE1CBBNSZjrwHTx2LrM9LbXbGH7gb9qLCYc3JUkLW883SeQ2H
+pgdbname=${TEST_DB}
+simnet=1
+rpclisten=127.0.0.1:17273
+debuglevel=trace
+regfeeconfirms=1
+signingkeypass=keypass
+pgpass=${PG_PASS}
+adminsrvon=1
+adminsrvpass=adminpass
+adminsrvaddr=127.0.0.1:16542
+EOF
+else
```

Comment:
Instead of duplicating the whole file in both branches of the if-else, let's just write the file without `pgpass` and then `echo pgpass=${PG_PASS} >> ./dcrdex.conf` if the `PG_PASS` is set (`if [[ -v PG_PASS ]]; then`).  I'm 100% fine with bash specific code in these scripts (prefer it even), but I wouldn't object to `-z ${PG_PASS}` either.

Reply:
got `zsh` as my default, it doesn't recognize `-v` here. I'm thinking og explicitly setting bash as the target, any objections?

Reply:
I suppose if `-v` is the only thing that doesn't work with zsh we might as well use `-z` instead to try to keep bashisms out.  I just didn't want to torture ourselves to support any shell when bash is the most common, but it looks like it's not a burden so far.